src/hotspot/os/aix/os_aix.cpp

```diff
@@ -619,66 +602,42 @@ void os::init_system_properties_values() {
-
-    unsigned long long virt_total = 0;
-    unsigned long long real_total = 0;
-    unsigned long long real_free = 0;
-    unsigned long long pgsp_total = 0;
-    unsigned long long pgsp_free = 0;
-    if (libo4::get_memory_info(&virt_total, &real_total, &real_free, &pgsp_total, &pgsp_free)) {
-      pmi->virt_total = virt_total;
-      pmi->real_total = real_total;
-      pmi->real_free = real_free;
-      pmi->pgsp_total = pgsp_total;
-      pmi->pgsp_free = pgsp_free;
-      return true;
-    }
+  // On AIX, I use the (dynamically loaded) perfstat library to retrieve memory statistics
+  // See:
+  // http://publib.boulder.ibm.com/infocenter/systems/index.jsp
+  //        ?topic=/com.ibm.aix.basetechref/doc/basetrf1/perfstat_memtot.htm
+  // http://publib.boulder.ibm.com/infocenter/systems/index.jsp
+  //        ?topic=/com.ibm.aix.files/doc/aixfiles/libperfstat.h.htm
+
+  perfstat_memory_total_t psmt;
+  memset (&psmt, '\0', sizeof(psmt));
+  const int rc = libperfstat::perfstat_memory_total(nullptr, &psmt, sizeof(psmt), 1);
+  if (rc == -1) {
+    trcVerbose("perfstat_memory_total() failed (errno=%d)", errno);
+    assert(0, "perfstat_memory_total() failed");
     return false;
+  }
 
-  } else {
-
-    // On AIX, I use the (dynamically loaded) perfstat library to retrieve memory statistics
-    // See:
-    // http://publib.boulder.ibm.com/infocenter/systems/index.jsp
-    //        ?topic=/com.ibm.aix.basetechref/doc/basetrf1/perfstat_memtot.htm
-    // http://publib.boulder.ibm.com/infocenter/systems/index.jsp
-    //        ?topic=/com.ibm.aix.files/doc/aixfiles/libperfstat.h.htm
-
-    perfstat_memory_total_t psmt;
-    memset (&psmt, '\0', sizeof(psmt));
-    const int rc = libperfstat::perfstat_memory_total(nullptr, &psmt, sizeof(psmt), 1);
-    if (rc == -1) {
-      trcVerbose("perfstat_memory_total() failed (errno=%d)", errno);
-      assert(0, "perfstat_memory_total() failed");
-      return false;
-    }
-
-    assert(rc == 1, "perfstat_memory_total() - weird return code");
-
-    // excerpt from
-    // http://publib.boulder.ibm.com/infocenter/systems/index.jsp
-    //        ?topic=/com.ibm.aix.files/doc/aixfiles/libperfstat.h.htm
-    // The fields of perfstat_memory_total_t:
-    // u_longlong_t virt_total         Total virtual memory (in 4 KB pages).
-    // u_longlong_t real_total         Total real memory (in 4 KB pages).
-    // u_longlong_t real_free          Free real memory (in 4 KB pages).
-    // u_longlong_t pgsp_total         Total paging space (in 4 KB pages).
-    // u_longlong_t pgsp_free          Free paging space (in 4 KB pages).
+  assert(rc == 1, "perfstat_memory_total() - weird return code");
 
-    pmi->virt_total = psmt.virt_total * 4096;
-    pmi->real_total = psmt.real_total * 4096;
-    pmi->real_free = psmt.real_free * 4096;
-    pmi->pgsp_total = psmt.pgsp_total * 4096;
-    pmi->pgsp_free = psmt.pgsp_free * 4096;
+  // excerpt from
+  // http://publib.boulder.ibm.com/infocenter/systems/index.jsp
```

Comment:
Same here. URL on one line. Does it still work?

Reply:
The link is dead, it does not work any more. I think we can remove it.